tests/baselines/reference/falsinessCallExpressionCoercion.errors.txt

```diff
@@ -0,0 +1,124 @@
+
+
+==== falsinessCallExpressionCoercion.ts (6 errors) ====
+    function test1() {
+        function canAccess() { return false; }
+    
+        if (!canAccess) { // error
+             ~~~~~~~~~
+!!! error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
+        }
+    }
+    
+    function test2() {
+        function canAccess() { return false; }
+    
+        if (!canAccess) { // ok
+        }
+    
+        canAccess();
+    }
+    
+    function test3() {
+        function canAccess() { return false; }
+    
+        if (!!!canAccess) { // ok
+        }
+    }
+    
+    function test4(canAccess: () => boolean) {
+        if (!canAccess) { // error
+             ~~~~~~~~~
+!!! error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
+        }
+    }
+    
+    function test5(canAccess?: () => boolean) {
+        if (!canAccess) { // ok
+        }
+    }
+    
+    function test6() {
+        const x = {
+            foo: {
+                bar() { return true; }
+            }
+        };
+    
+        if (!x.foo.bar) { // error
+             ~~~~~~~~~
+!!! error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
+        }
+    }
+    
+    function test7() {
+        const x = {
+            foo: {
+                bar() { return true; }
+            }
+        };
+    
+        if (!x.foo.bar) { // ok
+        }
+    
+        x.foo.bar();
+    }
+    
+    class Test8 {
+        maybeIsUser?: () => boolean;
+    
+        isUser() {
+            return true;
+        }
+    
+        test() {
+            if (!this.isUser) { // error
+                 ~~~~~~~~~~~
+!!! error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
+            }
+    
+            if (!this.maybeIsUser) { // ok
+            }
+        }
+    }
+    
+    class Test9 {
+        isUser() {
+            return true;
+        }
+    
+        test() {
+            if (!this.isUser) { // ok
+            }
+    
+            this.isUser();
+        }
+    }
+    
+    function test10() {
+        function canAccess() { return false; }
+    
+        const res = canAccess
+        if (!res) { // error
+             ~~~
+!!! error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
+        }
+    }
+    
+    function test11() {
+        function canAccess() { return false; }
+    
+        if (!canAccess) { // ok
+             ~~~~~~~~~
+!!! error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
+        } else {
+            canAccess()

```

Comment:
This is erroring, and it seems that `forEachChild` is not finding this node

But if I change the if to `if (!canAccess || true)` then it does find this call

Any idea why it's sometimes ignoring the `else`?

Reply:
Actually

```ts
    const item = () => true;

    if (!item || true) {
    } else {
    }
```
doesn't fail when it should so might be a different cause

The issue is though that 

```ts
    const item = () => true;

    if (!item) {
    } else {
      item
    }
```

fails when it shouldn't, whereas

```ts
    const item = () => true;

    if (!item) {
    } else {
    }

    item
```

is fine

Reply:
nevermind figured this one out